### ISSUE TITLE
Features to add

### ISSUE THREAD

**Issue by kingbri1** [theroyallab/YALS#3]
Now:
- [x] Passing of model/context params
- [x] Auth permission endpoints
- [x] GPU splitting
- [x] Token metrics on generate
- [x] Stop reasons
- [x] Tokenizer encode/decode endpoints
- [x] Argparser that links to config model

Future/Minor:
- [x] Batching
- [ ] Logprobs
- [x] Extra samplers not in lcpp (ex. min_tokens)
- [x] JSON schemas
- [x] Sampler overrides
- [x] Inline model loading

**Comment by emzaedu** [comment]
> Now:
> 
>     * [x]  Passing of model/context params[x]  Auth permission endpoints[x]  GPU splitting[x]  Token metrics on generate[x]  Stop reasons[x]  Tokenizer encode/decode endpoints[x]  Argparser that links to config model
> 
> 
> Future/Minor:
> 
>     * [ ]  Batching[ ]  Logprobs[ ]  Extra samplers not in lcpp (ex. min_tokens)[ ]  JSON schemas[ ]  Sampler overrides[ ]  Inline model loading

Support for Draft models will be great

**Comment by fizzAI** [comment]
> Extra samplers not in lcpp (ex. min_tokens)

I was working on this in my own lcpp wrapper, it might well be easier to move sampling out of lcpp into the JS layer (ie, take raw logits from lcpp then do sampling/final softmax in JS)
It would make it much easier to extend too ^-^

**Comment by CoffeeVampir3** [comment]
The sampling time is actually quite significant, for small models sampling can actually dominate the computing costs. There's other issues, sampling outside of native doesn't parallelize well since you now need synchronization between the sampler step. I don't find this design to have any redeeming qualities, prototyping samplers is not one of the primary goals.

**Comment by kingbri1** [comment]
Sampler overrides added in 19c0e26baf484fef1e7aacf3518abdf5a4afe150

**Comment by CoffeeVampir3** [comment]
- Batching
- Min Tokens
- LLGuidance JSON Schemas

Are now supported in the bindings for bindings-new.

**Comment by Ph0rk0z** [comment]
How about vision since it's sorely missing from official llama.cpp

I know.. big ask. Right now only on ollama and kobold.cpp

**Comment by kingbri1** [comment]
Inline model loading added, starting from 123d2cecd2897ccbc2588e8c365139f8840ead9d

**Comment by Ph0rk0z** [comment]
One thing I'm missing is sampler order. Yes, there is temperature last but it's not fully the same. Can eek out much more coherence using llama.cpp. Can trim low tokens with min_P, then do top_K to limit more and finally do stuff like dry/xtc on a much better set of tokens than defaults.

Top sigma is missing since it inherits Tabby stuff in sillytavern but I can just work around it.

**Comment by kingbri1** [comment]
Closing this issue because everything has been implemented besides logprobs and that'll take some time.
